GiniVision/Classes/GiniPDFDocument.swift

```diff
@@ -56,6 +57,48 @@ final public class GiniPDFDocument: NSObject, GiniVisionDocument {
         }
     }
     
+    fileprivate func renderFirstPage(fromPdf pdf:CGPDFDocument) -> UIImage? {
+        var pdfImage:UIImage?
+        let pdfDoc = pdf
+        
+        if let pdfPage:CGPDFPage = pdfDoc.page(at: 1) {
+            let pageRect:CGRect = normalizedRect(forBoxRect: pdfPage.getBoxRect(.cropBox), withRotationAngle: pdfPage.rotationAngle)
+            
+            // Create context
+            UIGraphicsBeginImageContextWithOptions(CGSize(width: pageRect.width, height: pageRect.height), false, 0.0)
+            let context:CGContext = UIGraphicsGetCurrentContext()!
+            
+            // Fill context color
+            context.setFillColor(UIColor.white.cgColor)
+            context.fill(pageRect)
+            
+            // Align PDF's cropBox to the context
+            context.translateBy(x: 0, y: pageRect.size.height)
+            context.scaleBy(x: 1, y: -1)
+            context.concatenate(pdfPage.getDrawingTransform(.cropBox, rect: pageRect, rotate: 0, preserveAspectRatio: true))
```

Comment:
Will that infamous pdf (with it's page rotated 270°) be rendered correctly?

Reply:
Yes, It will work as expected (different orientations will also work). ☺️